src/lib/AutoControlledComponent.js

```diff
@@ -185,11 +197,13 @@ export default class AutoControlledComponent extends Component {
       // warn about failed attempts to setState for keys not listed in autoControlledProps
       const illegalKeys = _.difference(_.keys(maybeState), autoControlledProps)
       if (!_.isEmpty(illegalKeys)) {
-        console.error([
-          `${name} called trySetState() with controlled props: "${illegalKeys}".`,
-          'State will not be set.',
-          'Only props in static autoControlledProps will be set on state.',
-        ].join(' '))
+        console.error(
+          [
+            `${name} called trySetState() with controlled props: "${illegalKeys}".`,
+            'State will not be set.',
+            'Only props in static autoControlledProps will be set on state.',
+          ].join(' '),
+        )
```

Comment:
We cannot simply rename `componentWillReceiveProps()` to `componentDidUpdate()` there because we will have an additional `render()` after `setState()`, just compare

```
componentWillReceiveProps() 
render() 
// vs
render() 
componentDidUpdate 
render() 
componentDidUpdate 
```

Example: https://codesandbox.io/s/4x35y87rl4

---

The right way is to use there [`getDerivedStateFromProps()`](https://reactjs.org/docs/react-component.html#static-getderivedstatefromprops). This will be more complicated refactor :(